sycl/plugins/esimd_cpu/esimd_libcm.hpp

```diff
@@ -0,0 +1,327 @@
+//==------------ esimd_libcm.hpp libcm interface definitions --------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ESIMD_LIBCM_HPP_
+#define _ESIMD_LIBCM_HPP_
+
+#ifndef __SYCL_DEVICE_ONLY__
+
+#include <CL/sycl/backend_types.hpp>
+#include <CL/sycl/detail/accessor_impl.hpp>
+#include <CL/sycl/detail/common.hpp>
+#include <CL/sycl/detail/export.hpp>
+#include <CL/sycl/detail/helpers.hpp>
+#include <CL/sycl/detail/host_profiling_info.hpp>
+#include <CL/sycl/detail/kernel_desc.hpp>
+#include <CL/sycl/detail/type_traits.hpp>
+#include <CL/sycl/group.hpp>
+#include <CL/sycl/id.hpp>
+#include <CL/sycl/kernel.hpp>
+#include <CL/sycl/nd_item.hpp>
+#include <CL/sycl/range.hpp>
+
+#include <esimdcpu_runtime.h>
+
+#include <functional>
+#include <thread>
+#include <vector>
+
+#define LAMBDA_WRAPPER_TMPL(ARGTYPE, TAG)                                      \
+  typedef std::function<void(const ARGTYPE &)> LambdaFunction_##TAG;           \
+                                                                               \
+  extern "C" struct LambdaWrapper_##TAG {                                      \
+    LambdaFunction_##TAG Func;                                                 \
+    std::vector<ARGTYPE> ArgVector;                                            \
+  };                                                                           \
+                                                                               \
+  template <typename LambdaTy>                                                 \
+  auto makeWrapper_##TAG(std::vector<ARGTYPE> &ArgVector, LambdaTy F) {        \
+    std::unique_ptr<LambdaWrapper_##TAG> Wrapper(new LambdaWrapper_##TAG);     \
+    Wrapper->Func = LambdaFunction_##TAG(F);                                   \
+    Wrapper->ArgVector = ArgVector;                                            \
```

Comment:
this is one more extreme inefficiency, which should be fixed in this PR I believe. The huge vector of the iteration range size is now copied. It should be either reference, or `runIterationSpace`/`run` should fill it in directly

Reply:
ArgVector is removed.